ui/selectors/selectors.js

```diff
@@ -705,7 +714,7 @@ export function getMetaMaskCachedBalances(state, networkChainId) {
   if (state.metamask.accountsByChainId?.[chainId]) {
     return Object.entries(state.metamask.accountsByChainId[chainId]).reduce(
       (accumulator, [key, value]) => {
-        accumulator[key] = value.balance;
+        accumulator[key.toLowerCase()] = value.balance;
```

Comment:
Keeping the keys lowercase in the result object so that it is backwards compatible.